apps/src/lab2/views/dialogs/ShareDialog.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useCallback, useState} from 'react';
+import Typography from '@cdo/apps/componentLibrary/typography';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import FocusLock from 'react-focus-lock';
+import moduleStyles from './ShareDialog.module.scss';
+import QRCode from 'qrcode.react';
+import copyToClipboard from '@cdo/apps/util/copyToClipboard';
+import {hideShareDialog} from '@cdo/apps/code-studio/components/shareDialogRedux';
+
+const CopyToClipboardButton: React.FunctionComponent<{shareUrl: string}> = ({
+  shareUrl,
+}) => {
+  const [copiedToClipboard, setCopiedToClipboard] = useState(false);
+
+  const handleCopyToClipboard = () => {
+    copyToClipboard(shareUrl, () => {
+      setCopiedToClipboard(true);
+    });
+  };
+
+  return (
+    <button
+      type="button"
+      onClick={handleCopyToClipboard}
+      className={moduleStyles.copyToClipboard}
+    >
+      <FontAwesomeV6Icon
+        iconName={copiedToClipboard ? 'clipboard-check' : 'clipboard'}
+        iconStyle="thin"
+        className={moduleStyles.copyToClipboardIcon}
+      />
+      Copy link to project
+    </button>
+  );
+};
+
+const ShareDialog: React.FunctionComponent<{shareUrl: string}> = ({
+  shareUrl,
+}) => {
+  const dispatch = useAppDispatch();
+
+  const handleClose = useCallback(
+    () => dispatch(hideShareDialog()),
+    [dispatch]
+  );
+
+  return (
+    <FocusLock>
+      <div className={moduleStyles.dialogContainer}>
+        <div id="share-dialog" className={moduleStyles.shareDialog}>
+          <Typography semanticTag="h1" visualAppearance="heading-lg">
+            Share Project
```

Comment:
loc? I'm guessing we have a localized string for this already?

Reply:
I was going to hold off on loc since we are expecting a design pass on this dialog soon, but both strings are already available, so will do so!